pilot/pkg/model/push_context.go

```diff
@@ -1480,6 +1486,10 @@ func (ps *PushContext) initVirtualServices(env *Environment) error {
 	ps.virtualServiceIndex.privateByNamespaceAndGateway = map[string]map[string][]config.Config{}
 	ps.virtualServiceIndex.publicByGateway = map[string][]config.Config{}
 
+	if features.FilterGatewayClusterConfig {
+		ps.virtualServiceIndex.destinationsByGateway = make(map[string]map[string]sets.Set[int])
```

Comment:
why `sets.Set[int]`? I don't see us using the port value anywhere?

Reply:
> is it feasible to re-use virtualServiceDestinations instead of copy?

Because `virtualServiceDestinations` returns `map[string]sets.Set[int]`, if we want to keep hosts only, we will have to copy the keys out

So we have 3 options:
1. extract destinations directly from vs
2. reuse `virtualServiceDestinations`
3. reuse `virtualServiceDestinations` and copy hosts out

Which one would you prefer?